File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/main/MainResponse.java
package fr.myprysm.vertx.elasticsearch.action.main;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@DataObject(generateConverter = true)
public class MainResponse {

    private String nodeName;
    private String version;
    private String clusterName;
    private String clusterUuid;
    private String build;
    private boolean available;

    public MainResponse(JsonObject json) {
        MainResponseConverter.fromJson(json, this);
    }

    public JsonObject toJson() {
        JsonObject json = new JsonObject();
        MainResponseConverter.toJson(this, json);
        return json;
    }

}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/main/MainResponse.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.main

import fr.myprysm.vertx.elasticsearch.action.main.MainResponse

fun MainResponse(
  available: Boolean? = null,
  build: String? = null,
  clusterName: String? = null,
  clusterUuid: String? = null,
  nodeName: String? = null,
  version: String? = null): MainResponse = fr.myprysm.vertx.elasticsearch.action.main.MainResponse().apply {

  if (available != null) {
    this.setAvailable(available)
  }
  if (build != null) {
    this.setBuild(build)
  }
  if (clusterName != null) {
    this.setClusterName(clusterName)
  }
  if (clusterUuid != null) {
    this.setClusterUuid(clusterUuid)
  }
  if (nodeName != null) {
    this.setNodeName(nodeName)
  }
  if (version != null) {
    this.setVersion(version)
  }
}

